controllers/serviceclaim_controller.go

```diff
@@ -80,18 +80,26 @@ func (r *ServiceClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	rsName, err := r.processServiceClaim(ctx, req, rsl, sclaim)
+	rs, err := r.processServiceClaim(ctx, req, rsl, sclaim)
 	if err != nil {
 		l.Error(err, "unable while processing ServiceClaim")
 		return ctrl.Result{}, err
 	}
 	sclaim.Status.State = "Resolved"
 	sclaim.Status.ClaimID = uuid.New().String()
-	sclaim.Status.RegisteredService = rsName
+	sclaim.Status.RegisteredService = rs.Name
 	if err := r.Status().Update(ctx, &sclaim); err != nil {
 		l.Error(err, "unable to update the ServiceClaim", "ServiceClaim", sclaim)
 		return ctrl.Result{}, err
 	}
+
+	// Update RegisteredService status
+	rs.Status.State = primazaiov1alpha1.RegisteredServiceStateClaimed
+	if err := r.Status().Update(ctx, &rs); err != nil {
```

Comment:
if we fail here, the registered service may be claimed from another ServiceClaim. I'd update the RegisteredService  status before the ServiceClaim's one.
If we proceed this way, we should also check if in a previous reconciliation an error occurred while updating ServiceClaim's status. This could be done by adding a property into the RegisteredService's status referencing to the binded Claim (e.g., `ClaimRef`). When reconciling the service class we should check if exists a RegisteredService with this field populated to the current ServiceClaim. In this case we update the ServiceClaim status again.